lib/std/builtin.zig

```diff
@@ -160,54 +160,340 @@ pub const OptimizeMode = enum {
+    powerpc64_elf: CommonOptions,
+    powerpc64_elf_altivec: CommonOptions,
+    powerpc64_elf_v2: CommonOptions,
+
+    // Calling conventions for the `powerpc` and `powerpcle` architectures.
+    powerpc_sysv: CommonOptions,
+    powerpc_sysv_altivec: CommonOptions,
+    powerpc_aix: CommonOptions,
+    powerpc_aix_altivec: CommonOptions,
+
+    /// The standard `wasm32`/`wasm64` calling convention, as specified in the WebAssembly Tool Conventions.
+    wasm_watc: CommonOptions,
+
+    /// The standard `arc` calling convention.
+    arc_sysv: CommonOptions,
+
+    // Calling conventions for the `avr` architecture.
+    avr_gnu,
+    avr_builtin,
+    avr_signal,
+    avr_interrupt,
+
+    /// The standard `bpfel`/`bpfeb` calling convention.
+    bpf_std: CommonOptions,
+
+    // Calling conventions for the `csky` architecture.
+    csky_sysv: CommonOptions,
+    csky_interrupt: CommonOptions,
+
+    // Calling conventions for the `hexagon` architecture.
+    hexagon_sysv: CommonOptions,
+    hexagon_sysv_hvx: CommonOptions,
+
+    /// The standard `lanai` calling convention.
+    lanai_sysv: CommonOptions,
+
+    /// The standard `loongarch64` calling convention.
+    loongarch64_lp64: CommonOptions,
+
+    /// The standard `loongarch32` calling convention.
+    loongarch32_ilp32: CommonOptions,
+
+    // Calling conventions for the `m68k` architecture.
+    m68k_sysv: CommonOptions,
+    m68k_gnu: CommonOptions,
+    m68k_rtd: CommonOptions,
+    m68k_interrupt: CommonOptions,
+
+    /// The standard `msp430` calling convention.
+    msp430_eabi: CommonOptions,
+
+    /// The standard `propeller1` calling convention.
+    propeller1_sysv: CommonOptions,
+
+    /// The standard `propeller2` calling convention.
+    propeller2_sysv: CommonOptions,
+
+    // Calling conventions for the `s390x` architecture.
+    s390x_sysv: CommonOptions,
+    s390x_sysv_vx: CommonOptions,
+
+    /// The standard `ve` calling convention.
+    ve_sysv: CommonOptions,
+
+    // Calling conventions for the `xcore` architecture.
+    xcore_xs1: CommonOptions,
+    xcore_xs2: CommonOptions,
+
+    // Calling conventions for the `xtensa` architecture.
+    xtensa_call0: CommonOptions,
+    xtensa_windowed: CommonOptions,
+
+    // Calling conventions for the `amdgcn` architecture.
+    amdgcn_device: CommonOptions,
+    amdgcn_kernel,
+    amdgcn_cs: CommonOptions,
+
+    // Calling conventions for the `nvptx` architecture.
+    nvptx_device,
+    nvptx_kernel,
+
+    // Calling conventions for kernels and shaders on the `spirv`, `spirv32`, and `spirv64` architectures.
+    spirv_device,
+    spirv_kernel,
+    spirv_fragment,
+    spirv_vertex,
+
+    /// Options shared across most calling conventions.
+    pub const CommonOptions = struct {
+        /// The boundary the stack is aligned to when the function is called.
+        /// `null` means the default for this calling convention.
+        incoming_stack_alignment: ?u64 = null,
```

Comment:
If this refers to the stack alignment right _before_ the caller executes its `call` instruction, I think "before the function is called" from #21209 was better phrasing than "when the function is called."

Reply:
I'm leaving this as-is for now as a slight cop-out because I'm not sure whether it's necessarily "before the function is called" on *all* targets.